Dyninst::SymtabAPI::Module::getAllVariables missing?

Reported by Laksono Adhianto to @mxz297 and @hainest.
---

I am trying to gather the list of variables (name, size and the offset address):

```
static void dump_variables(Module *module) {
  vector<Variable*> variables;
  if (module->getAllVariables(variables)) {
    for(auto  i=0; i<variables.size(); i++) {
        Variable *var = variables[i];
        printf("\t%d. %p %x  %d\n", i, var->mangled_names_begin(), var->getOffset(), var->getSize());
    }
  }
}

vector<Module*> modules;
if (syms->getAllModules(modules)) {
  for(auto i=0; i<modules.size(); i++) {
          Module *module = modules[i];
          dump_variables(module);
  }
}
```

This works nice before, but failed when linked with SymtabAPI in the master branch:

```
g++ -I /tmp/pkgs/dyninst//include/ -g -c dvar.cpp
g++ -L /tmp/pkgs/dyninst//lib/ -lboost_system -lsymtabAPI -ldyninstAPI -lcommon dvar.o -o dvar
dvar.o: In function `dump_variables(Dyninst::SymtabAPI::Module*)':
/home/la5/git/hpctoolkit-tests/dyninst/variables/dvar.cpp:30:    undefined reference to `Dyninst::SymtabAPI::Module::getAllVariables(std::vector<Dyninst::SymtabAPI::Variable*, std::allocator<Dyninst::SymtabAPI::Variable*> >&)'
collect2: error: ld returned 1 exit status
Makefile:8: recipe for target 'dvar' failed
make: *** [dvar] Error 1
```

It seems Dyninst::SymtabAPI::Module::getAllVariables is not implemented anymore ? I looked at the source code, the closest API is Symtab::getAllVariables, or (to some extent) mapped_object::getAllVariables:

```
401 bool Symtab::getAllVariables(std::vector<Variable *> &ret)
402 {
403     ret = everyVariable;
404     return (ret.size() > 0);
405 }
```

```
137 class mapped_object : public codeRange, public Dyninst::PatchAPI::DynObject {
238     bool getAllVariables(pdvector<int_variable *> &vars);
```

Can you check if this Dyninst::SymtabAPI::Module::getAllVariables is really deprecated or accidentally missing?

Source code: [dvar.tar.gz.zip](https://github.com/dyninst/dyninst/files/4697601/dvar.tar.gz.zip)

--- a/symtabAPI/h/Module.h
+++ b/symtabAPI/h/Module.h
@@ -189,8 +189,6 @@ namespace Dyninst{
 									 NameType nameType = anyName,
 									 bool isRegex = false,
 									 bool checkCase = true);
-			bool getAllVariables(std::vector<Variable *> &ret);
-
 
 			// Type output methods
 			virtual bool findType(boost::shared_ptr<Type>& type, std::string name);

